functests/vrf/vrf.go

```diff
@@ -0,0 +1,338 @@
+	return defIntName
+}
+
+func getOverlapIP(cs *apiclient.ClientSet, nodeName string, podName string) string {
+	tempPodDefinition := defineHostNetworkdPod(podName, TestNamespace, nodeName, false)
+	err := cs.Create(context.Background(), tempPodDefinition)
+	Expect(err).ToNot(HaveOccurred())
+	Eventually(func() k8sv1.PodPhase {
+		tempPod, _ := cs.Pods(TestNamespace).Get(context.Background(), tempPodDefinition.Name, metav1.GetOptions{})
+		return tempPod.Status.Phase
+	}, podWaitingTime, time.Second).Should(Equal(k8sv1.PodRunning))
+
+	pod, err := cs.Pods(TestNamespace).Get(context.Background(), tempPodDefinition.Name, metav1.GetOptions{})
+	Expect(err).ToNot(HaveOccurred())
+	return pod.Status.PodIP
+}
+
+func waitUntilPodCreatedAndRunning(cs *apiclient.ClientSet, podStruct *k8sv1.Pod) {
+	err := cs.Create(context.Background(), podStruct)
+	Expect(err).ToNot(HaveOccurred())
+	Eventually(func() k8sv1.PodPhase {
+		tempPod, _ := cs.Pods(TestNamespace).Get(context.Background(), podStruct.Name, metav1.GetOptions{})
+		return tempPod.Status.Phase
+	}, podWaitingTime, time.Second).Should(Equal(k8sv1.PodRunning))
+}
+
+func podHasCorrectVrfConfig(cs *apiclient.ClientSet, podName string, vrfMapsConfig []map[string]string) error {
+	pod, err := cs.Pods(TestNamespace).Get(context.Background(), podName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	for _, vrfMapConfig := range vrfMapsConfig {
+		validateVrfIPAddrCommand := []string{"ip", "addr", "show", fmt.Sprintf("%s", vrfMapConfig["vrfInterface"])}
+		Eventually(func() bool {
+			vrfIface, _ := pods.ExecCommand(cs, *pod, validateVrfIPAddrCommand)
+			return strings.Contains(vrfIface.String(), vrfMapConfig["vrfClinetIP"])
+		}, podWaitingTime, 5*time.Second).Should(BeTrue(), fmt.Errorf("VRF interface is not present"))
+
+		validateVRFRouteTableCommand := []string{"ip", "route", "show", "vrf", fmt.Sprintf("%s", vrfMapConfig["vrfName"])}
+		Eventually(func() bool {
+			vrfRouteTable, _ := pods.ExecCommand(cs, *pod, validateVRFRouteTableCommand)
+			return strings.Contains(vrfRouteTable.String(), vrfMapConfig["vrfClinetIP"])
+		}, podWaitingTime, 5*time.Second).Should(BeTrue(), fmt.Errorf(fmt.Sprintf("VRF %s route table is not present", vrfMapConfig["vrfName"])))
+	}
+	return nil
+}
+
+func redefineAsPrivileged(pod *k8sv1.Pod) *k8sv1.Pod {
+	pod.Spec.Containers[0].SecurityContext = &k8sv1.SecurityContext{}
+	b := true
+	pod.Spec.Containers[0].SecurityContext.Privileged = &b
+	return pod
+}
+
+func newVRFTestParameters(Node string, IPStack string) (*vrfTestParameters, error) {
+	VRFTestParameters := new(vrfTestParameters)
+	err := validateSrtParam(Node, nodeParameters)
+	if err != nil {
+		return nil, err
+	}
+	VRFTestParameters.Node = Node
+
+	err = validateSrtParam(IPStack, ipStackParameters)
+	if err != nil {
+		return nil, err
+	}
+	VRFTestParameters.IPStack = IPStack
+	return VRFTestParameters, nil
+}
+
+func validateSrtParam(param string, paramRange []string) error {
```

Comment:
if you use it only once, can you use the loop directly?

Reply:
I use it twice in newVRFTestParameters